Projects/RemoteCameraRobot/remote_robot.py

```diff
@@ -0,0 +1,252 @@
+def page(page_name):
+    return render_template("{}".format(page_name))
+
+@app.route("/static/<path:path>")
+def send_static(path):
+    return send_from_directory(directory_path, path)
+
+#############################
+### Video Streaming Stuff ###
+#############################
+
+class StreamingOutput(object):
+    '''
+    Class to which the video output is written to.
+    The buffer of this class is then read by StreamingHandler continuously.
+    '''
+    def __init__(self):
+        self.frame = None
+        self.buffer = io.BytesIO()
+        self.condition = Condition()
+
+    def write(self, buf):
+        if buf.startswith(b'\xff\xd8'):
+            # New frame, copy the existing buffer's content and notify all
+            # clients it's available
+            self.buffer.truncate()
+            with self.condition:
+                self.frame = self.buffer.getvalue()
+                self.condition.notify_all()
+            self.buffer.seek(0)
+        return self.buffer.write(buf)
+
+class StreamingHandler(server.BaseHTTPRequestHandler):
+    '''
+    Implementing GET request for the video stream.
+    '''
+    def do_GET(self):
+        if self.path == '/stream.mjpg':
+            self.send_response(200)
+            self.send_header('Age', 0)
+            self.send_header('Cache-Control', 'no-cache, private')
+            self.send_header('Pragma', 'no-cache')
+            self.send_header('Content-Type', 'multipart/x-mixed-replace; boundary=FRAME')
+            self.end_headers()
+            try:
+                while True:
+                    with output.condition:
+                        output.condition.wait()
+                        frame = output.frame
+                    self.wfile.write(b'--FRAME\r\n')
+                    self.send_header('Content-Type', 'image/jpeg')
+                    self.send_header('Content-Length', len(frame))
+                    self.end_headers()
+                    self.wfile.write(frame)
+                    self.wfile.write(b'\r\n')
+            except Exception as e:
+                logging.warning(
+                    'Removed streaming client %s: %s',
+                    self.client_address, str(e))
+        else:
+            self.send_error(404)
+            self.end_headers()
+
+class StreamingServer(socketserver.ThreadingMixIn, server.HTTPServer):
+    allow_reuse_address = True
+    daemon_threads = True
+
+#############################
+### Aggregating all calls ###
+#############################
+
+if __name__ == "__main__":
+    # registering both types of signals
+    signal.signal(signal.SIGINT, signal_handler)
+    signal.signal(signal.SIGTERM, signal_handler)
+
+    # firing up the video camera (pi camera)
+    camera = picamera.PiCamera(resolution='320x240', framerate=30)
+    output = StreamingOutput()
+    camera.start_recording(output, format='mjpeg')
+    logging.info("Started recording with picamera")
+    STREAM_PORT = 5001
```

Comment:
Do we have a place in documentation where we keep a list of all the ports? We should.